src/constraints.jl

```diff
@@ -934,10 +934,10 @@ function all_constraints(
     # TODO: Support JuMP's set helpers like SecondOrderCone().
     f_type = moi_function_type(function_type)
     if set_type <: MOI.AbstractScalarSet
-        constraint_ref_type = ConstraintRef{Model, _MOICON{f_type, set_type},
+        constraint_ref_type = ConstraintRef{<:AbstractModel, _MOICON{f_type, set_type},
```

Comment:
This needs to match the type of `model`. Either change it back to `Model`, or add a type parameter for `model`.